cmd/scan/scan.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"strings"
+
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcutil"
+)
+
+type NewBTCAddress struct {
+	Addresses []string `json:"btc_addresses""`
+}
+
+type Address struct {
+	Addr         string `json:"address"`
+	MinScanBlock int64  `json:"min_scan_block"`
+	MidScanBlock int64  `json:"mid_scan_block"`
+	MaxScanBlock int64  `json:"max_scan_block"`
+	Txs          []Tx   `json:"txs"`
+}
+
+type Tx struct {
+	TxHash      string `json:"tx_hash"`
+	BlockHash   string `json:"block_hash"`
+	ParentHash  string `json:"parent_hash"`
+	BlockHeight int64  `json:"block_height"`
+}
+
+type Deposit struct {
+	Addr string
+	Tx   Tx
+}
+
+func ScanBlock(client *rpcclient.Client, blockID int64) ([]Deposit, error) {
+	blockHash, err := client.GetBlockHash(blockID)
+	if err != nil {
+		return nil, err
+	}
+	block, err := client.GetBlockVerboseTx(blockHash)
+	if err != nil {
+		return nil, err
+	}
+	parentHash := block.PreviousHash
+	var deposits []Deposit
+	var depTx Tx
+	depTx.BlockHeight = blockID
+	depTx.BlockHash = blockHash.String()
+	depTx.ParentHash = parentHash
+	for _, tx := range block.RawTx {
+		for i, addr := range tx.Vout {
+			depTx.TxHash = fmt.Sprintf("%s:%d", tx.Txid, i)
+
+			if len(addr.ScriptPubKey.Addresses) > 0 {
+				deposits = append(deposits, Deposit{
+					Addr: addr.ScriptPubKey.Addresses[0],
```

Comment:
What does it mean if there is more than one address in here

Reply:
there is one address always, can be situation when "Unable to decode output address", and this if prevent it. Look an example transaction with hash:
500f2be2ebd5193578de1e521fbd7a3d9849a8983a70a8bef89e3d0a89a9a18b

Reply:
I replaced it on "if addr.ScriptPubKey.Addresses != nil "

Reply:
Do you have the docs for this API call/data?

Reply:
nope, i just test manually this moment